File: X's and zeros.cpp
#include <iostream> 
auto show_playing_filed(char arr[9]){ //вывод игрового поля в консоль
         for (int i = 0; i < 7;i = i + 3) {
            std::cout << 
            arr[i] << " | " << arr[i + 1] << " | " << arr[i + 2]
            << std::endl;
            if (i < 6) std::cout 
                << "---------"
                << std::endl; 
    }
}

int main(){
    system("chcp 65001 > nul"); //изменение кодировки в консоли
    char 
        Background_of_playing_field[9] = //с англ.Фон игрового поля 
        {' ', ' ', ' ', //изначальный вид игрового поля
        ' ', ' ', ' ',
        ' ', ' ', ' '}, 
        current_course = 'x'; //с англ. текущий ход. 
    int combinations[8][3] = //комбинации для победы
        {{0, 1, 2},
        {3, 4, 5},
        {6, 7, 8}, 
        {0, 3, 6}, 
        {1, 4, 7}, 
        {2, 5, 8}, 
        {0, 4, 8}, 
        {2, 4, 6}},
        motion, // c англ. ход
        counter; // с англ. счётчик 

    while (true) {
        counter = 0; //сброс счётчика
        std::cout << current_course << " ходят\n\n";
        show_playing_filed(Background_of_playing_field);
        std::cout << "\nХоди\n >";
        std::cin >> motion; //игрок выбирает клетку, куда будет ходить
        system("cls");
        if ((std::cin.fail()) || ((motion > 9) || (motion < 1))){ //конструкция на случай, если пользователь ввёл не то
            std::cin.clear();
            std::cin.ignore();
            std::cout << "Внимание! Вводите только номер клетки, куда собираетесь ходить (начиная с 1 и заканчивая 9)" << std::endl;
            continue;
        }
        if (Background_of_playing_field[motion - 1] != '0' & //проверка на то, не занята ли клетка
        Background_of_playing_field[motion - 1] != 'x') {
            Background_of_playing_field[motion - 1] = current_course;
        }
        else{
            std::cout << "Кто то уже сделал ход сюда(" <<  Background_of_playing_field[motion - 1] << ")" << std::endl;     
            continue;
        }
        for (int i = 0; i < 8; ++i){
                if (Background_of_playing_field[combinations[i][0]] == current_course & //Проверка условий победы
                Background_of_playing_field[combinations[i][1]] == current_course & 
                Background_of_playing_field[combinations[i][2]] == current_course ){
                    std::cout << "Игрок " << current_course << " победил!\n";
                    show_playing_filed(Background_of_playing_field);
                    system("pause");
                    return 0;
                }
        }
        for (int i = 0; i < 9; ++i){ //проверка на перегрузку 
            if(Background_of_playing_field[i] != ' '){
                counter++;
            }
            if (counter == 9){ //условия перегрузки
                std::cout << "Поле перегруженно! Ничья!" << std::endl;
                show_playing_filed(Background_of_playing_field);
                system("pause");
                return 0;
            }
        }
        if (current_course =='x'){ //рокировка
            current_course = '0';
        }
        else{
            current_course = 'x';
        }
    }
} 

File: X's and zeros.rb
def playing_field(arr) #Вывод игрового поля на экран. С англ. Игровое поле
  for i in 0..8
    if i == 1 or i == 4 or i == 7
      print " | "  
      print arr[i]  
      print " | "
    else
      print arr[i]  
    end
    if i == 2 or i == 5
            puts "\n-----------"
    end 
  end
end
#основной код
Background_of_playing_field = ["1","2","3", "4","5","6", "7","8","9"] #с англ.Фон игрового поля 
player = "x" #иксы ходят первыми(условно)
counter = 0 #счётчик для обнаружения перегру3ки
while true
  counter = 0 #сброс счётчика
  playing_field(Background_of_playing_field) #Вывод игрового поля на экран
  print "\n", player, " ходят \n", "> " 
  players_turn = (gets.to_i) - 1 #с англ. ход игрока
  system("cls")   #очистка экрана
  if ((Background_of_playing_field[players_turn] != "x") and (Background_of_playing_field[players_turn] != "0")) 
    Background_of_playing_field[players_turn] = player #присвоение 3аднику 3начения
  else
    puts "Клетка 3анята!"
    redo
  end
 
  if (((Background_of_playing_field[0] == player) and (Background_of_playing_field[1] == player) and (Background_of_playing_field[2] == player)) or #Проверка условий победа
    ((Background_of_playing_field[3] == player) and (Background_of_playing_field[4] == player) and (Background_of_playing_field[5] == player)) or
    ((Background_of_playing_field[6] == player) and (Background_of_playing_field[7] == player) and (Background_of_playing_field[8] == player)) or 
    ((Background_of_playing_field[0] == player) and (Background_of_playing_field[3] == player) and (Background_of_playing_field[6] == player)) or
    ((Background_of_playing_field[1] == player) and (Background_of_playing_field[4] == player) and (Background_of_playing_field[7] == player)) or
    ((Background_of_playing_field[2] == player) and (Background_of_playing_field[5] == player) and (Background_of_playing_field[8] == player)) or
    ((Background_of_playing_field[0] == player) and (Background_of_playing_field[4] == player) and (Background_of_playing_field[8] == player)) or
    ((Background_of_playing_field[6] == player) and (Background_of_playing_field[4] == player) and (Background_of_playing_field[2] == player)) )
    print player, " победил!\n"
    playing_field(Background_of_playing_field) #Вывод игрового поля на экран
    puts " "
    system("pause")
    exit
  end
  for i in 0..8 #проверка на перегру3ку
      if   (Background_of_playing_field[i] == "x") or  (Background_of_playing_field[i] == "0")
        counter += 1
      end
      if counter == 9 
        puts "Поле перегруженно! Ничья!"  
        playing_field(Background_of_playing_field) #Вывод игрового поля на экран
        puts " "
        system("pause")
        exit
      end
  end
  if player == "x" #смена игрока
    player = "0"
  else
    player = "x"
  end
end
